src/main/java/sirius/web/controller/BasicController.java

```diff
@@ -161,15 +163,72 @@ public void fail(WebContext ctx, HandledException error) {
     }
 
     @Override
-    public void onError(WebContext ctx, HandledException error) {
-        if (ctx.isResponseCommitted() || defaultRoute == null) {
-            fail(ctx, error);
+    public void onError(WebContext webContext, HandledException error) {
+        if (webContext.isResponseCommitted() || defaultRoute == null) {
+            fail(webContext, error);
```

Comment:
update fail to also handle the ErrorCodeException accordingly

Reply:
fail now checks for an ErrorCodeException